components/raftstore/src/store/fsm/store.rs

```diff
@@ -150,12 +150,19 @@ impl StoreMeta {
     }
 }
 
-pub struct RaftRouter<S: Snapshot> {
-    pub router: BatchRouter<PeerFsm<S>, StoreFsm>,
+pub struct RaftRouter<EK, ER, S>
```

Comment:
Could we use `Ek::Snapshot` instead of `S`?

Reply:
+1

Reply:
Yes. That is part of the intended future work. I will start on that patch now, but ask that you consider landing this first.